From 6ebe199c6f092b9482162445690fc760dca31f07 Mon Sep 17 00:00:00 2001
From: Wendy Liang <wendy.liang@xilinx.com>
Date: Thu, 24 Sep 2020 11:36:03 -0700
Subject: [PATCH 1662/1852] misc: xilinx-ai-engine: Add errors categories masks
 macro

commit 4b59a32c39929a04003b88ac9c4cdeb04a99d2fd from
https://github.com/Xilinx/linux-xlnx.git

Add AI engine errors categories masks macro for other modules to check
what categories are with the categories bitmap returned by the get
categories function.

Signed-off-by: Wendy Liang <wendy.liang@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 include/linux/xlnx-ai-engine.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/include/linux/xlnx-ai-engine.h b/include/linux/xlnx-ai-engine.h
index 9f489305238e..ff71b2bb03af 100644
--- a/include/linux/xlnx-ai-engine.h
+++ b/include/linux/xlnx-ai-engine.h
@@ -28,6 +28,19 @@
 #define AIE_ERROR_CATEGORY_DMA			8U
 #define AIE_ERROR_CATEGORY_MEM_PARITY		9U
 
+/* AIE error category bit mask */
+#define AIE_ERROR_CATMASK(c)			BIT(AIE_ERROR_CATEGORY_##c)
+#define AIE_ERROR_CATEGORY_MASK_SATURATION	AIE_ERROR_CATMASK(SATURATION)
+#define AIE_ERROR_CATEGORY_MASK_FP		AIE_ERROR_CATMASK(FP)
+#define AIE_ERROR_CATEGORY_MASK_STREAM		AIE_ERROR_CATMASK(STREAM)
+#define AIE_ERROR_CATEGORY_MASK_ACCESS		AIE_ERROR_CATMASK(ACCESS)
+#define AIE_ERROR_CATEGORY_MASK_BUS		AIE_ERROR_CATMASK(BUS)
+#define AIE_ERROR_CATEGORY_MASK_INSTRUCTION	AIE_ERROR_CATMASK(INSTRUCTION)
+#define AIE_ERROR_CATEGORY_MASK_ECC		AIE_ERROR_CATMASK(ECC)
+#define AIE_ERROR_CATEGORY_MASK_LOCK		AIE_ERROR_CATMASK(LOCK)
+#define AIE_ERROR_CATEGORY_MASK_DMA		AIE_ERROR_CATMASK(DMA)
+#define AIE_ERROR_CATEGORY_MASK_MEM_PARITY	AIE_ERROR_CATMASK(MEM_PARITY)
+
 /**
  * enum aie_module_type - identifies different hardware modules within a
  *			  tile type. AIE tile may have memory and core
-- 
2.31.1

